### DISCUSSION TITLE
How do I create an enchantment?

### DISCUSSION THREAD

**Discussion by zi2ven** [FabricMC/fabric-api#3878] [FAQ] [answered]
# Version
Minecraft: 1.21
Fabric Loom: 1.7.1
Yarn Mappings=1.21+build.2
Loader Version=0.15.11
Fabric Kotlin Version=1.11.0+kotlin.2.0.0
# Problem
I try to create an enchantment like Wiki does 
```java
public class FrostEnchantment extends Enchantment {
 
    public FrostEnchantment() {
        super(Enchantment.Rarity.UNCOMMON, EnchantmentTarget.WEAPON, new EquipmentSlot[] {EquipmentSlot.MAINHAND});
    }
}
```
But I don't have the class EnchantmentTarget, and Enchantment.Rarity.UNCOMMON is actually incorrect, it should be Rarity.UNCOMMON, and this constructor is also problematic

I want to know what should I do?

**Comment by Jark5455** [comment, marked as answer]
Are you using 1.21? Now enchantments are data-driven and have to be created through datapacks or registries

**Reply by zi2ven** [reply]
Really?But how can i bind my events to my enchantments?
Can you give me an example or a Wiki link?

**Comment by FlintLeng** [comment]
## Re: Platform-aware DriftLock anchors

Great idea! Platform-aware anchors would significantly improve the reliability of agent-based mining workflows.

**Potential approach:**

1. **Architecture detection**: Use platform.system() and platform.machine() to detect OS and CPU architecture at runtime
2. **Anchor tier selection**: Route to different DriftLock endpoints based on detected platform
3. **Fallback chain**: If primary anchor is unreachable, automatically fall back to secondary tier

**Concerns worth addressing:**

- DriftLock relies on the assumption that a given hardware fingerprint maps to a single platform tier
- The latency between platform detection and anchor selection matters for time-sensitive mining operations
- Cross-platform compatibility of the agent template itself

**Implementation suggestion:**

Consider a layered approach: platform detection at the template level, DriftLock anchor selection at the agent runtime level. This separates concerns cleanly and makes the system more testable.

Would love to see this integrated with the existing bounty system — agents could self-classify into hardware tiers automatically.
